File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/C/P06-BirgeVieta.cpp
// Real roots of a polynomial
#include <stdio.h>
#include "roots.h"

int main()
{
   const int n = 4;                                   // degree of polynomial
   double ax[] = {1e0, 0e0, -5e0, 0e0, 4e0};                  // coefficients
   double xx[n+1];                                                   // zeros
   int i, nx;

   BirgeVieta(ax,n,xx,nx);                                  // nx zeros found
   
   for (i=1; i<=nx; i++) printf("x[%1i] = %10.5f\n",i,xx[i]);
   if (nx <= 0) printf(" Non-existing real zeros\n");
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/Python/P06-BirgeVieta.py
# Real roots of a polynomial by the Birge-Vieta method
from math import *
from roots import *

n = 4                                                  # degree of polynomial
ax = [1e0, 0e0, -5e0, 0e0, 4e0]                                # coefficients
xx = [0]*(n+1)                                                        # zeros

nx = BirgeVieta(ax,n,xx)                                     # nx zeros found

for i in range(1,nx+1): print("x[",i,"] = ",xx[i])
if (nx <= 0): print(" Non-existing real zeros")
